src/main.rs

```diff
@@ -93,7 +95,10 @@ async fn main() -> Result<()> {
     create_test_project(&client).await?;
 
     let mut actors = Vec::new();
-    let mut error_futures = FuturesUnordered::new();
+    let mut error_futures: FuturesUnordered<
+        Pin<Box<dyn Future<Output = Option<anyhow::Error>>>>,
```

Comment:
For my edification: was this needed to keep things type-checking when you added the disk module, or is this a more idiomatic way to collect futures that I didn't know about?

Reply:
This was my probably naive to keep things type-checking :) For some reason it didn't like collecting the instance and disk related futures together without this.